### ISSUE TITLE
[Website Request] Bangumi

### ISSUE THREAD

**Issue by liurui39660** [MALSync/MALSync#792]
## Website
- Name: Bangumi
- Link: https://bgm.tv/ or https://bangumi.tv/
- Language(s): Simplified Chinese

## Other details
This is an anime/manga progress management site, resembles MAL and AniList. It has [official APIs](https://github.com/bangumi/api) provided, but some features are not implemented yet (e.g. fetching all watched records of a user). I hope this can be added to MAL's progress sync.

**Comment by lolamtisch** [comment]
Yeah without user lists not really a good fit for malsync. And specially the episode based watching states will be difficult too, as the specials are includeded there too.

**Comment by liurui39660** [comment]
Is crawling the web page an option? In the user profile, there will be 24 records per page, the item id is in the HTML `id`. After that, there are APIs to interact with individual items (fetch details, fetch progress, set status, etc).

I can help open an issue on their side if you can help list required APIs.

**Comment by lolamtisch** [comment]
We dont want to do any session bassed web crawling anymore. Another problem is that many Features work on the malid and i did not see them saving it. The experience will be a lot lot worse without it. The search and mapping to the correct entry will be very bad in comparision and you lose all next episode times and notifications and some other things.

**Comment by liurui39660** [comment]
I see the difficulty. For the mapping, I have no idea how MAL Sync works, but I have a script that uses the Japanese name provided by Bangumi to search the corresponding item in AniList. The accuracy is about 95%, but Bangumi merged some items, like the main story and OVAs at the same season.

Still, if you can provide a list of required APIs, I can help open an issue and ask them.

**Comment by lolamtisch** [comment]
This should prety much be all informations needed per api call.
```
Single Entry
    internalId: any
    malId: number
    type: 'anime' | 'manga'
    url: string
    userStatus: 'watching' | 'pantowatch' | 'completed' ....
    userScore: any
    userWatchedEpisode / chapter: number
    userWatchedVolumens: number
    totalEpisodes: number
    totalVolumes: number
    image: string
    globalScore: any

list
    internalId: any
    malId: number
    type: 'anime' | 'manga'
    title: string
    url: string
    totalEp: number
    userWatchedEpisode / chapter: number
    userStatus: 'watching' | 'pantowatch' | 'completed' ....
    userScore: any
    image: string
    
Search
    id: number;
    name: string;
    altNames: string[];
    url: string;
    image: string;
    
Overview metadata:
   Just metadata not important
```

**Comment by lolamtisch** [comment]
List should be able to be filtered by userState.
And if they add malid finding the entry using the malId would be needed too

**Comment by louisroyer** [comment]
> I see the difficulty. For the mapping, I have no idea how MAL Sync works, but I have a script that uses the Japanese name provided by Bangumi to search the corresponding item in AniList. The accuracy is about 95%, but Bangumi merged some items, like the main story and OVAs at the same season.
> 
> Still, if you can provide a list of required APIs, I can help open an issue and ask them.

Maybe you could rely on [Wikidata](https://www.wikidata.org) to do the mapping, or at least as to better evaluate the accuracy of your script. There is an [identifier for bangumi](https://www.wikidata.org/wiki/Property:P5732) and an [identifier for MAL](https://www.wikidata.org/wiki/Property:P4086). It is far from complete for the moment ([1523 items with both bangumi id and mal id](https://w.wiki/5oTn)), but it could be improved. Here are [some tools](https://www.wikidata.org/wiki/Wikidata:Tools/For_programmers) allowing to perform queries ;  [wikidatanpm](https://www.npmjs.com/package/wikidatanpm) may interest you since it is a nodejs module.

**Comment by gh-zj978** [comment]
Any news? ( No pressure, just curious :3 )
